### ISSUE TITLE
Extend assembly language with additional integer representations and definitions

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-vm#633]
# Feature request

1. Additional integer representations

Miden supports bitwise operations for the u32 "datatype", these operations are typically used with bitmasks, supporting syntax for it can make the assembly easier to read. E.g.:

```
push.0b11 u32checked_and
```

2. Definitions

In some contexts it makes sense to use constants, e.g. value `-1` can be defined as:

```
push.0 sub.1
```

Which evaluates to `18446744069414584320`. Having a directive to name this value, e.g. `negative_one`, may make it easier to read the assembly.

**Comment by bobbinth** [comment]
Agreed on the first one - being able to specify values in binary form could be helpful.

The second makes sense too, but it is a bit tricky as `-1` could mean different things as when interpreted as a field element vs. a `u32` value. I think we could do something like specifying the type of a value like `felt` or `u32`.

More generally, I think naming values (i.e.,. declaring constants) could be super useful and probably deserves its own issue. For example, something like:
```
const.FOO.10

being
    push.FOO # pushes 10 onto the stack
end
```
Though, I don't love the `const.NAME.value` syntax. Maybe we can come up with something better without complicating the parser too much.

**Comment by hackaugusto** [comment]
Created the issue for const vales: https://github.com/0xPolygonMiden/miden-vm/issues/634
